modules/seqexec/engine/src/main/scala/seqexec/engine/Engine.scala

```diff
@@ -22,15 +21,14 @@ import org.log4s.getLogger
 
 import scala.concurrent.ExecutionContext
 
-class Engine[D: ActionMetadataGenerator, U](implicit ev: ActionMetadataGenerator[D]) {
+class Engine[D, U](stateL: Lens[D, Engine.State]) {
```

Comment:
I basically turned around the relation between Engine.State and the state used by the upper level.